Doc/library/urllib.request.rst

```diff
@@ -1236,6 +1236,38 @@ Use of Basic HTTP Authentication::
    urllib.request.install_opener(opener)
    urllib.request.urlopen('http://www.example.com/login.html')
 
+.. note::
+
+   Normally, when you first try to access to a protected page, the server should
+   send back a ``401 Unauthorized`` response, with a ``WWW-Authenticate`` header,
+   containing the authentication scheme in use, and the realm of the page
+   (e.g., ``WWW-Authenticate: Basic realm="Python Documentation"``). After what,
+   you can perform a new request with a ``Authorization`` header, containing
+   your credentials encoded in base64 (e.g., ``Authorization: Basic <CREDENTIALS>``).
+
+   This is what :class:`HTTPBasicAuthHandler` is automatically doing for you by default.
+
+However, when the server doesn't properly implement Basic HTTP Authentication, you should
+instead use :class:`HTTPBasicAuthHandler` with :class:`HTTPPasswordMgrWithPriorAuth`::
+
+   import urllib.request
+
+   # Set-up an authentication handler.
+   pwd_mgr = urllib.request.HTTPPasswordMgrWithPriorAuth()
+   pwd_mgr.add_password(
+      realm=None,  # No realm
```

Comment:
I think comment is redundant here

Reply:
I expanded the comment, to give a bit more context 👍 

I added comments to explain what keyword argument is exactly doing what 😃